src/interaction/sessionrecording.cpp

```diff
@@ -0,0 +1,1207 @@
+    //_externInteract.timeInteraction(pbFrame);
+    addKeyframe(pbFrame._timestamp, pbFrame);
+}
+
+void SessionRecording::playbackScript() {
+    double timeRef;
+    double timeOs, timeRec, timeSim;
+    unsigned int numScriptLines;
+    datamessagestructures::ScriptMessage pbFrame;
+
+    if (isDataModeBinary()) {
+        readFromPlayback(timeOs);
+        readFromPlayback(timeRec);
+        readFromPlayback(timeSim);
+        try {
+            readFromPlayback(pbFrame._script);
+        }
+        catch (std::bad_alloc&) {
+            LERROR(fmt::format("Allocation error with script playback from keyframe entry {}",
+                _playbackLineNum - 1));
+            return;
+        }
+
+        if (!_playbackFile) {
+            LERROR(fmt::format("Error reading script playback from keyframe entry {}",
+                _playbackLineNum - 1));
+            return;
+        }
+    } else {
+        std::istringstream iss(_playbackLineParsing);
+        std::string entryType;
+        std::string tmpReadbackScript;
+
+        iss >> entryType;
+        iss >> timeOs >> timeRec >> timeSim;
+        iss >> numScriptLines;
+        std::getline(iss, tmpReadbackScript); //iss >> tmpReadbackScript;
+        pbFrame._script.append(tmpReadbackScript);
+        if (iss.fail()) {
+            LERROR(fmt::format("Error parsing script line {} of playback file", _playbackLineNum));
+            return;
+        } else if (!iss.eof()) {
+            LERROR(fmt::format("Did not find an EOL at line {} of playback file", _playbackLineNum));
+            return;
+        }
+        if (numScriptLines > 1) {
+            //Now loop to read any subsequent lines if is a multi-line script
+            for (unsigned int i = 1; i < numScriptLines; ++i) {
+                pbFrame._script.append("\n");
+                std::getline(_playbackFile, tmpReadbackScript);
+                pbFrame._script.append(tmpReadbackScript);
+            }
+        }
+    }
+    timeRef = getAppropriateTimestamp(timeOs, timeRec, timeSim);
+    //timeRef = getEquivalentSimulationTime(timeOs, timeRec, timeSim);
+
+    //Call script scheduler with this new script entry
+    //std::string timeDescription = SpiceManager::ref().dateFromEphemerisTime(timeRef, "YYYY MON DD HR:MN:SC.###");
+    //ghoul::Dictionary scriptDict(ghoul::Dictionary{ {KeyTime, timeDescription },
+    //                                                {KeyForwardScript, pbFrame._script} }
+    //                            );
+    //global::scriptScheduler.loadScripts({ { "1", scriptDict } });
+    addKeyframe(timeRef, pbFrame._script);
+}
+
+void SessionRecording::addKeyframe(double timestamp, interaction::KeyframeNavigator::CameraPose keyframe)
+{
+    unsigned int indexIntoCameraKeyframesFromMainTimeline = static_cast<unsigned int>(_keyframesCamera.size());
+    _keyframesCamera.push_back(keyframe);
+    _timeline.push_back({ recordedType::camera,
```

Comment:
Can do a `std::move` on the `keyframe`

Indentation:

```
_timeline.push_back({
    recordedType::camera,
    indexIntoCameraKeyframesFromMainTimeline,
    timestamp
});

Reply:
Done